### PR TITLE
An implementation of MINRES-QLP

### PR DESCRIPTION
**Pull request by amontoison**
For the moment, this `MINRES-QLP` isn't optimized for singular inconsistent systems.
We could easily add conditions for returning the minimum-norm solution.

For the same amount of memory and few more operations,  this `MINRES-QLP` is a robust alternative of `MINRES`.

### CODE REVIEW

File: src/minres_qlp.jl [hunk 1]
```diff
@@ -226,13 +229,20 @@ function minres_qlp(A :: AbstractLinearOperator, b :: AbstractVector{T};
...
     rNorm = abs(ζbarₖ₊₁)
     push!(rNorms, rNorm)
 
+    # Update ‖Arₖ₋₁‖ estimate
+    # ‖ Arₖ₋₁ ‖ = |ζbarₖ| * √((λₖ)² + (γbarₖ)²)
+    ArNorm = abs(ζbarₖ) * √(λₖ^2 + (cₖ * βₖ₊₁)^2)
+    push!(ArNorms, ArNorm)
+
     # Update stopping criterion.
+    iter == 1 && (ϟ = atol + rtol * ArNorm)
```
Comment (amontoison) [inline src/minres_qlp.jl]: @dpo Is it a relevant stopping condition ? I reused what we do with the residual norm, but I'm not sure that we should do the same thing with Aresidual norm...

Comment (dpo) [reply src/minres_qlp.jl]: What is that weird symbol? A lightning strike? Does it represent the current ArNorm?

Comment (amontoison) [reply src/minres_qlp.jl]: `\koppa`, a `k` in old grec. I found it when I wanted to write `\kappa`. It's a nice (grec) symbol. It represents `atol + rtol * Ar₀`. It's similar to `ε = atol + rtol * r₀`.

Comment (dpo) [reply src/minres_qlp.jl]: It's a fine stopping criterion, but I'm not crazy about that symbol. I like \kappa because it reminds us of "KKT".

Comment (amontoison) [reply src/minres_qlp.jl]: Ok, I just need to verify that `‖ Arₖ₋₁ ‖ = |ζbarₖ| * √((λₖ)² + (γbarₖ)²)` or `‖ Arₖ₋₁ ‖ = |ζbarₖ| * √((λbarₖ)² + (γbarₖ)²)` and change my lightning stike symbol.

Comment (amontoison) [reply src/minres_qlp.jl]: There was a mistake in `‖ Arₖ₋₁ ‖`, I corrected it. I finished what I wanted to do with `MINRES-QLP`.

**Comment by dpo** [comment]
🤘 🚀

**Pull request merged**